client/components/web-preview/component.jsx

```diff
@@ -0,0 +1,205 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { hasTouch } from 'lib/touch-detect';
+import { isMobile } from 'lib/viewport';
+import { localize } from 'i18n-calypso';
+import RootChild from 'components/root-child';
+import { setPreviewShowing } from 'state/ui/actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+import WebPreviewContent from './content';
+
+export class WebPreview extends PureComponent {
```

Comment:
The `WebPreview` class now does quite different things when `isContentOnly` is `true` or `false` and there is a lot of conditional logic.

We could simplify that by extracting a helper component:
```jsx
// component.jsx
import WebPreviewContent from './content';

class WebPreviewModal {
  // the original WebPreview component, with lifetime methods, event listeners,
  // rendering a modal dialog. Not changed at all.
}

// wrapper component that does the isContentOnly decisions
const WebPreview = props => {
  const { isContentOnly, ...restProps } = props;
  const WebPreviewComponent = isContentOnly
    ? WebPreviewContent
    : WebPreviewModal;
  return <WebPreviewComponent { ...restProps } />;
}

export default WebPreview;
```

Two minor unrelated nits that could be optionally fixed;
- the `PureComponent` is not needed on a `connect`-ed component. `connect()` already does all the necessary shallow props comparing.
- the `connect` for `WebPreview` binds a `recordTracksEvent` action, but seems to not use it at all. `WebPreviewContent` uses it, but it gets it from its own `connect`...